carmin.yaml

```diff
@@ -133,28 +332,147 @@ definitions:
       defaultStudy:
         type: string
+        format: ascii
       supportedAPIVersion:
         type: string
-  errorCodeAndMessage:
+        format: ascii
+    additionalProperties: true # allow extensions
+  Pipeline:
+    type: object
+    required:
+      - identifier
+      - name
+      - version
+      # - parameters optional in 0.2
+    properties:
+      identifier:
+        type: string
+        format: ascii
+      name:
+        type: string
+      version:
+        type: string
+        format: ascii
+      description:
+        type: string
+      canExecute:
+        type: boolean
+        description: true if the user who requested the pipeline can execute it
+      parameters:
+        type: array
+        items:
+          $ref: '#/definitions/PipelineParameter'
+      errorCodesAndMessages:
+        type: array
+        items:
+          $ref: '#/definitions/ErrorCodeAndMessage'
+  PipelineParameter:
+    type: object
+    required:
+      - name
+      - type
+      - isOptional
+      - isReturnedValue
+    properties:
+      name:
+        type: string
+        format: ascii
+      type:
+        $ref: '#/definitions/ParameterType'
+      isOptional:
+        type: boolean
+      isReturnedValue:
+        type: boolean
+      defaultValue:
+        description: Default value. It must be consistent with the parameter type.
+      description:
+        type: string
+  ParameterType:
+    type: string
+    enum: [File,String,Boolean,Int64,Double,List]
+  ErrorCodeAndMessage:
     type: object
     required:
       - code
       - message
     properties:
       code:
         type: integer
-        format: int32
+        format: int64
       message:
         type: string
-    
\ No newline at end of file
+  Execution:
+    type: object
+    required:
+      - name
+      - pipelineIdentifier
+      - inputValues
+    properties:
+      identifier:
+        type: string
+        format: ascii
+        readOnly: true
+        description: execution id. Must always be present in responses.
+      name:
+        type: string
+        description: execution name
+      pipelineIdentifier:
+        type: string
+        format: ascii
+        description: which pipeline that started this execution
+      timeout:
+        type: integer
+        format: int64
+        description: The timeout in seconds until which the execution is killed and deleted with all its files. 0 or absense means no timeout
+      status:
+        type: string
+        enum: [initializing,ready,running,finished,initialization_failed,execution_failed,unknown,killed]
+        readOnly: true
+        description: The status of the execution. Must always be present in responses.
+      inputValues:
+        type: object
+        additionalProperties: true
+        description:
+          Represents the input as a key/value object. The types should respect the parameters of the pipeline used for the execution.
+      returnedFiles:
```

Comment:
If we support directories, there must be a way for client code to check whether it is a file or directory before downloading the path (the downloading code will not be the same). This is not a problem for sftp because this possibility is in the protocol. I think this is more difficult for http/https links and I just do not know how to handle this case without changing the API (e.g. adding a is_directory boolean somewhere). This is for keeping API simple for most frequent use case (i.e. one output = one file) that I suggested to list all output files (but only files) for each output parameter.